src/VisualStudio/CSharp/Impl/LanguageService/CSharpCodeCleanupFixer.cs

```diff
@@ -0,0 +1,116 @@
+        {
+            switch(scope)
+            {
+                case TextBufferCodeCleanUpScope textBufferScope:
+                    return FixTextBufferAsync(textBufferScope, operationContext, cancellationToken);
+                case IVsHierarchyCodeCleanupScope hierarchyContentScope:
+                    return FixHierarchyContentAsync(hierarchyContentScope, cancellationToken);
+                default:
+                    return Task.FromResult(false);
+            }            
+        }
+
+        private Task<bool> FixHierarchyContentAsync(IVsHierarchyCodeCleanupScope hierarchyContent, CancellationToken cancellationToken)
+        {
+            // TODO: this one will be implemented later
+            // https://github.com/dotnet/roslyn/issues/30165
+            var hierarchy = hierarchyContent.Hierarchy;
+            if (hierarchy == null)
+            {
+                // solution
+                return Task.FromResult(false);
+            }
+
+            var itemId = hierarchyContent.ItemId;
+
+            if (hierarchy.GetCanonicalName(itemId, out var path) == 0)
+            {
+                var attr = File.GetAttributes(path);
+                if (attr.HasFlag(FileAttributes.Directory))
+                {
+                    // directory
+                }
+                else
+                {
+                    // document
+                }
+            }
+
+            return Task.FromResult(false);
+        }
+
+        private async Task<bool> FixTextBufferAsync(TextBufferCodeCleanUpScope textBufferScope, IUIThreadOperationContext operationContext, CancellationToken cancellationToken)
+        {
+            var buffer = textBufferScope.SubjectBuffer;
+            if (buffer != null)
+            {
+                using (var scope = operationContext?.AddScope(allowCancellation: true, description: EditorFeaturesResources.Applying_changes))
+                {
+
+                    var progressTracker = new ProgressTracker((description, completed, total) =>
+                    {
+                        if (scope != null)
+                        {
+                            scope.Description = description;
+                            scope.Progress.Report(new ProgressInfo(completed, total));
+                        }
+                    }
+                    );
+                    
+                    var document = buffer.CurrentSnapshot.GetOpenDocumentInCurrentContextWithChanges();
+                    var codeCleanupService = document.GetLanguageService<ICodeCleanupService>();
+
+                    // TODO: enable all diagnostics for now, need to be replace by inclusion/ exclusion list from .editorconfig
+                    // https://github.com/dotnet/roslyn/issues/30163
+                    var organizeUsingsSet = new OrganizeUsingsSet(true, true);
+                    var enabledDiagnostics = codeCleanupService.GetAllDiagnostics();
```

Comment:
is this @jinujoseph our decision to apply all? not just one that user choose?

Reply:
For now we apply all, we will have an exclusion list of diagnosticId in .editorconfig to let user exclude diagnosticIds that they don't want to fix
https://github.com/dotnet/roslyn/issues/30163

Reply:
yes , we will need a way to exclude this via .editorconfig for P1 .
Ultimately, this should apply only the options selected by user and that list will flow in from platform  API